components/homme/src/share/dimensions_mod.F90

```diff
@@ -4,13 +4,72 @@
 
 module dimensions_mod
 #ifdef CAM
+  use shr_kind_mod, only : r8=>shr_kind_r8
+#ifdef FVM_TRACERS
+  use constituents, only : ntrac_d=>pcnst ! _EXTERNAL
+#else
   use constituents, only : qsize_d=>pcnst ! _EXTERNAL
+#endif
 #endif
   implicit none
   private
 
+  integer, parameter             , public :: np = NP
+
 ! set MAX number of tracers.  actual number of tracers is a run time argument  
 #ifdef CAM
+#ifdef FVM_TRACERS
+  integer, parameter                      :: qsize_d =10 ! SE tracers (currently SE supports 10 condensate loading tracers)
+#else
+  integer, parameter                      :: ntrac_d = 0 ! No fvm tracers if CSLAM is off
+#endif
+  character(len=16),  allocatable, public :: cnst_name_gll(:)     ! constituent names for SE tracers
+  character(len=128), allocatable, public :: cnst_longname_gll(:) ! long name of SE tracers
+  logical                        , public :: lcp_moist = .true.
+
+  integer, parameter             , public :: nc = 3               ! cslam resolution
+  integer                        , public :: fv_nphys             ! physics-grid resolution - the "MAX" is so that the code compiles with NC=0
+  integer                        , public :: ntrac = 0            ! ntrac is set in dyn_comp
+  logical                        , public :: use_cslam = .false.  ! logical for CSLAM
```

Comment:
I see a bunch of constants in here that are not used anywhere in homme. Do you need them in other modules in CAM? Perhaps they should be defined there then...

Reply:
You are correct. Those variables are used by CAM. In the CAM-SE dycore, we have the same file `dimensions_mod.F90` to define those variables and we just copy them here for simplicity (also avoid changing many `use` statements if putting them in a different module). But it does seem unnecessary for E3SM and let me think how to handle it in a better way.

Reply:
Thanks @bartgol . This seems a good way to approach and I can move most of my changes to the CAM specific place instead. I will work on it now.